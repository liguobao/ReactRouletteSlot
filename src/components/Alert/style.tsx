/*
 * @Author: wzi
 * @Date: 2018-10-01 10:23:08
 * @Last Modified by: wzi
 * @Last Modified time: 2019-01-08 16:39:04
 */

import styled from '@common/helper/styled-component';

type WrapperProps = {
    mask: boolean;
};
export const Wrapper = styled.div`
    position: absolute;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    z-index: 100;
    background: ${(props: WrapperProps) => (props.mask ? '#988d8d' : null)};
    display: flex;
    align-items: center;
`;
export const ContextWrapper = styled.div`
    width: 100%;
    text-align: center;
`;
export const ContentWrapper = styled.span`
    border-radius: 10px;
    color: #fff;
    background-color: rgba(58, 58, 58, 0.9);
    display: inline-block;
    padding: 20px;
`;
export const Title = styled.div``;
export const Detail = styled.div``;
