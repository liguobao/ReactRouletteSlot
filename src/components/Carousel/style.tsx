/*
 * @Author: wzi
 * @Date: 2018-07-19 16:51:24
 * @Last Modified by: wzi
 * @Last Modified time: 2018-09-17 18:08:51
 */

import styled from '@common/helper/styled-component';
import { Carousel } from 'antd-mobile';
import CustomIcon from '@components/CustomIcon';

export const Img = styled(CustomIcon)`
    font-size: 100px;
`;
export const CarouselWrapper = styled(Carousel)`
    background: white;
    overflow: hidden;
`;
export const Wrapper = styled.div`
    min-height: 185px;
    transition: height 0.6s;

    ul {
        transition: height 0.6s;
        text-align: center;

        li a {
            min-height: 185px;
            img {
                width: 100%;
                max-width: 100%;
            }
        }
    }
`;

export const Holder = styled.div`
    height: 185px !important;
    text-align: center;
`;
