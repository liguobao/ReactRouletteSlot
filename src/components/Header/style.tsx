/*
 * @Author: wzi
 * @Date: 2018-09-18 10:58:02
 * @Last Modified by: wzi
 * @Last Modified time: 2018-09-18 17:16:58
 */

import styled from '@common/helper/styled-component';
import { NavBar, Icon } from 'antd-mobile';
// import CustomIcon from '@components/CustomIcon';

export const HeaderBar = styled(NavBar)``;

export const LeftIcon = styled(Icon).attrs({
    type: 'left',
})``;
