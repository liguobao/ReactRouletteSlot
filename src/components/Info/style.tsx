/*
 * @Author: wzi
 * @Date: 2018-09-17 17:42:15
 * @Last Modified by: wzi
 * @Last Modified time: 2018-09-30 11:27:17
 */

import styled from '@common/helper/styled-component';
import { Accordion, WhiteSpace, Flex } from 'antd-mobile';
import CustomIcon from '@components/CustomIcon';
export { WhiteSpace, Flex };
export const Info = styled(Accordion)``;
export const Panel = styled(Accordion.Panel)`
    font-size: 14px !important;
`;
export const Content = styled.div`
    padding: 15px;
`;
export const Icon = styled(CustomIcon).attrs({
    type: 'info',
})`
    font-size: 20px;
`;
