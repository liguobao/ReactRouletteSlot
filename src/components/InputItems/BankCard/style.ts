/*
 * @Author: wzi
 * @Date: 2018-09-26 18:28:36
 * @Last Modified by: wzi
 * @Last Modified time: 2018-09-30 16:09:04
 */
import styled from '@common/helper/styled-component';
import { Picker, List } from 'antd-mobile';
export const BankCardPicker = styled(Picker)``;
export const Item = styled(List.Item)`
    .am-list-extra {
        flex: 1;
    }
`;
