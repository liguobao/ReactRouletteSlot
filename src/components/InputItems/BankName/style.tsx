/*
 * @Author: wzi
 * @Date: 2018-09-26 18:28:36
 * @Last Modified by: wzi
 * @Last Modified time: 2018-09-29 16:19:44
 */
import styled from '@common/helper/styled-component';
import { Picker, List } from 'antd-mobile';
export const BankNamePicker = styled(Picker)``;
export const Item = styled(List.Item)`
    .am-list-extra {
        flex: 1;
    }
`;
