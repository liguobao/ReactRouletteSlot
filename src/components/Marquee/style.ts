/*
 * @Author: wzi
 * @Date: 2018-04-08 17:03:52
 * @Last Modified by: wzi
 * @Last Modified time: 2018-09-18 17:20:17
 */
import styled from '@common/helper/styled-component';
import CustomIcon from '@components/CustomIcon';
type LabelType = {
    type: 'pc' | 'mobile';
};
export const Notice = styled.span`
    vertical-align: super;
    margin-left: 1px;
`;
export const Label = styled.span`
    text-align: center;
    width: ${(props: LabelType) => (props.type === 'pc' ? '8%' : '20%')};
    background: #ed6b44;
    float: left;
    color: white;
    border-radius: 4px 0px 0px 4px;
`;

export const Horn = styled(CustomIcon)`
    margin-top: 0px;
`;
export const MalarqueeWrapper = styled.span`
    display: inline-block;
    width: ${(props: LabelType) => (props.type === 'pc' ? '90%' : '80%')};
`;
type MarqueeWrapperProp = {
    className?: string;
};
export const MarqueeWrapper = styled.div.attrs<MarqueeWrapperProp>({
    className: (props: MarqueeWrapperProp) => `${props.className} layout-width`,
})`
    background: white;
    border-radius: 5px;
`;
