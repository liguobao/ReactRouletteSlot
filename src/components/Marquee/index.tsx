/*
 * @Author: wzi
 * @Date: 2018-03-01 10:01:16
 * @Last Modified by: wzi
 * @Last Modified time: 2018-09-18 10:10:59
 */
import React from 'react';
import Malarquee from 'react-malarquee';
import { shouldUpdate } from '@common/decorator/decorator';
import errorBoundaryDecorator from '@common/decorator/errorBoundaryDecorator';
import { MarqueeWrapper, Label, Horn, Notice, MalarqueeWrapper } from './style';
type MarqueeProp = {
    content: string;
    fetchUrgentNotice: GlobalType.VoidFunc;
    className?: string;
    type?: 'pc' | 'mobile';
};
@shouldUpdate({
    props: ['content'],
})
@errorBoundaryDecorator('公告展示错误')
class Marquee extends React.Component<MarqueeProp> {
    componentDidMount() {
        this.fetch();
    }
    fetch = () => {
        const { content, fetchUrgentNotice } = this.props;
        if (!content.length) {
            fetchUrgentNotice();
        }
    };
    render() {
        const { content, type = 'pc', className } = this.props;
        return (
            <MarqueeWrapper className={className}>
                <Label type={type}>
                    <Horn type="horn" />
                    <Notice> 公告 : </Notice>
                </Label>
                <MalarqueeWrapper type={type}>
                    <Malarquee
                        fill={false}
                        hoverToPause
                        rate={50}
                        children={content || '暂无公告'}
                    />
                </MalarqueeWrapper>
            </MarqueeWrapper>
        );
    }
}
export default Marquee;
