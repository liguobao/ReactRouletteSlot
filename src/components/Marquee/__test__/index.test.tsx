/*
 * @Author: wzi
 * @Date: 2018-01-31 16:40:02
 * @Last Modified by: wzi
 * @Last Modified time: 2018-04-27 22:08:09
 */
import React from 'react';
import Marquee from '@components/Marquee';
import { shallow } from 'enzyme';
import 'jest-styled-components';
import toJson from 'enzyme-to-json';
import {  } from '../style';
describe('UserInfo', () => {
    let _props, _spies, _wrapper;
    beforeEach(() => {
        _spies = {};
        _props = {
            content: '哈哈哈',
            type: 'pc',
            fetchUrgentNotice: (_spies.fetchUrgentNotice = jest.fn(() => {
                console.log('更新紧急公告');
            })),
        };
        _wrapper = shallow(<Marquee {..._props} />);
    });
    test('整体', () => {
        expect(toJson(_wrapper)).toMatchSnapshot();
        expect(_spies.fetchUrgentNotice).toHaveBeenCalledTimes(0);
    });
    test('无内容时, 自动发出请求', () => {
        _wrapper = shallow(<Marquee {..._props} content={''} />);
        expect(_spies.fetchUrgentNotice).toHaveBeenCalledTimes(1);
    });
});
