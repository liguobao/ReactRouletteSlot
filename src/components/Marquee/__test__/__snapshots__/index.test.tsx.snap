// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserInfo 整体 1`] = `
<styled.div>
  <styled.span
    type="pc"
  >
    <Styled(Styled(CustomIcon))
      type="horn"
    />
    <styled.span>
       公告 : 
    </styled.span>
  </styled.span>
  <styled.span
    type="pc"
  >
    <Marquee
      fill={false}
      hoverToPause={true}
      rate={50}
    >
      哈哈哈
    </Marquee>
  </styled.span>
</styled.div>
`;
